libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -12,4 +13,19 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       _applicationId: applicationId,
     });
   }
+
+  async create(data: Partial<IntegrationEntity>): Promise<IntegrationEntity> {
+    const test = await this.find({
```

Comment:
```suggestion
    const test = await this.findOne({
```
This will be faster because it won't continue scanning after returning the first doc.